k8s/controllers/actions/catalogeval_controller.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	federationv1 "gopls-workspace/apis/federation/v1"
+
+	"gopls-workspace/utils/diagnostic"
+
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// CatalogReconciler reconciles a Site object
+type CatalogEvalReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	// ApiClient is the client for Symphony API
+	ApiClient utils.ApiClient
+}
+
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=federation.symphony,resources=catalogevalexpression/finalizers,verbs=update
+
+func (r *CatalogEvalReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ret ctrl.Result, reterr error) {
+	log := ctrllog.FromContext(ctx)
+	diagnostic.InfoWithCtx(log, ctx, "Entering the CatalogEvalExpression reconciler")
+
+	evalCR := &federationv1.CatalogEvalExpression{}
+
+	err := r.Get(ctx, req.NamespacedName, evalCR)
+	if err != nil {
+		diagnostic.ErrorWithCtx(log, ctx, err, "failed to parse CatalogEvalExpression")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	operationId := evalCR.GetOperationID()
+
+	// Clean up the cr if it is older than 10 hrs
+	if CheckNeedtoDelete(evalCR.ObjectMeta) {
+		// Delete the CR
+		diagnostic.InfoWithCtx(log, ctx, "Deleting the CatalogEvalExpression as it is older than 10 hrs", "CR", req.NamespacedName)
+		if err := r.Delete(ctx, evalCR); err != nil {
```

Comment:
Will delete call enter the reconcile loop again? I didn't see you remove the finalizer. Seems when you create you didn't use finalizer to lock the object.